### ISSUE TITLE
Parse error with escaped heredocs

### ISSUE THREAD

**Issue by sodapopcan** [rrrene/credo#1030]
* Credo version: 1.6.7-ref.main.07c2558+uncommittedchanges
* Erlang/Elixir version: 1.14.3
* Operating system: MacOS 13.1 (22C65)

### What were you trying to do?

There seem to be a few reports around heredocs that have been fixed.  I was easily able to work around this, and the "work around" is better than what I had, but credo actually crashes in the following situation:

I have a test to ensure a function removes all comments, but leave comments inside docstrings, so I had this in my setup:

```elixir
code = """
# I'm a comment
# and I continue on
# with commenting
@doc """
## Examples
# Comment in docstring
\"""
def some_code(a_fn_with_args) do
  # I'm commenting again why not
  do_the_thing_to(a_fn_with_args)
end
"""
```

### Expected outcome

`$ mix credo --strict` runs without crashing.

### Actual outcome

```bash
warning: Unexpected error while parsing test/path/to/test_test.exs: [%Credo.Issue{check: Credo.Code.ParserError, category: :error, priority: 0, severity: nil, message: "unexpected token: ", filename: "nofilename", line_no: [line: 15, column: 7], column: nil, exit_status: 0, trigger: nil, diff_marker: nil, meta: [], scope: nil}]
  lib/credo/check/readability/string_sigils.ex:53: Credo.Check.Readability.StringSigils.run/2
  lib/credo/check/readability/string_sigils.ex:5: Credo.Check.Readability.StringSigils.do_run_on_source_file/3
  (elixir 1.14.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
  (elixir 1.14.3) lib/task/supervised.ex:34: Task.Supervised.reply/4
  (stdlib 4.0.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

This is really no big deal as I just changed the outer heredoc delimiters to `'''` and got rid of the escaped line which is far nicer anyway, but I thought I would report the issue.

Also, this doesn't happen when run without the `--strict` option.

Thanks!

**Comment by rrrene** [comment]
I'm not sure if this is a Credo problem.

If you change

```elixir
@doc """
```

to

```elixir
@doc \"""
```

the warning should disappear (note the backslash).

**Comment by rrrene** [comment]
Closing this as this seemed to be resulting from a typo and be a compiler error due to an error in the provided code. 

If this is actually still a problem, please re-open.
